src/main/java/io/fabric8/jenkins/openshiftsync/ConfigMapWatcher.java

```diff
@@ -82,21 +82,19 @@ public void doRun() {
                             resourceVersion = configMaps.getMetadata()
                                     .getResourceVersion();
                         }
-                        synchronized(ConfigMapWatcher.this) {
-                            if (watches.get(namespace) == null) {
-                                logger.info("creating ConfigMap watch for namespace "
-                                        + namespace
-                                        + " and resource version "
-                                        + resourceVersion);
-                                watches.put(
-                                        namespace,
-                                        getAuthenticatedOpenShiftClient()
-                                        .configMaps()
-                                        .inNamespace(namespace)
-                                        .withResourceVersion(
-                                                resourceVersion)
-                                                .watch(new WatcherCallback<ConfigMap>(ConfigMapWatcher.this,namespace)));
-                            }
+                        if (watches.get(namespace) == null) {
+                            logger.info("creating ConfigMap watch for namespace "
+                                    + namespace
+                                    + " and resource version "
+                                    + resourceVersion);
+                            watches.put(
```

Comment:
In theory, this should be `watches.putIfAbsent(` since there is a gap between the get and put calls ... but since there should only be one lister thread running we should be OK.

That said, employing the proper pattern would be good in case folks copy this pattern in other scenarios where they cannot assume single thread.

So, yes, please make that change @mrsiano